File: app/src/main/java/com/hapi/aoptest/MainActivity.java
package com.hapi.aoptest;

import android.app.Activity;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;

import com.hapi.aopbeat.MethodBeatMonitorJava;
import com.hapi.mylibrary.EmpActivity;


public class MainActivity extends AppCompatActivity {

    View login ;
    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        verifyStoragePermissions(this);
        MethodBeatMonitorJava.checkDeep();

        setContentView(R.layout.activity_login);
        login = findViewById(R.id.login);
        login.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(MainActivity.this, EmpActivity.class);
                startActivity(i);
            }
        });
        new S().b();
    }

    private static final int REQUEST_EXTERNAL_STORAGE = 1;
    private static String[] PERMISSIONS_STORAGE = {
            "android.permission.READ_EXTERNAL_STORAGE",
            "android.permission.WRITE_EXTERNAL_STORAGE" };


    public  void verifyStoragePermissions(Activity activity) {

        try {
            //检测是否有写的权限
            int permission = ActivityCompat.checkSelfPermission(activity,
                    "android.permission.WRITE_EXTERNAL_STORAGE");
            if (permission != PackageManager.PERMISSION_GRANTED) {
                // 没有写的权限，去申请写的权限，会弹出对话框
                ActivityCompat.requestPermissions(activity, PERMISSIONS_STORAGE,REQUEST_EXTERNAL_STORAGE);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
    //在onCreate()方法中调用该方法即可

}

File: app/src/main/java/com/hapi/aoptest/MainActivity.kt
package com.hapi.aoptest

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.Choreographer

class MainActivity : AppCompatActivity() {

    var last = 0L
    private val fpsFrameCallback = Choreographer.FrameCallback {
        Log.d("fpsFrameCallback", "请求一阵   "+(it/1000000-last))
        last = it/1000000;

        onFrame(it)
    }

    private val choreographer by lazy { Choreographer.getInstance() }
    private fun onFrame(frameTimeNanos:Long){

            choreographer.postFrameCallback(fpsFrameCallback)
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        choreographer.postFrameCallback(fpsFrameCallback)
    }
}
